src/lib/openapi/mappers/mapper.ts

```diff
@@ -0,0 +1,5 @@
+export interface Mapper<PUBLIC, INTERNAL, I = Partial<PUBLIC>> {
+    fromPublic(input: PUBLIC): INTERNAL;
+    toPublic(input: INTERNAL): PUBLIC;
+    mapInput?(input: I): INTERNAL;
+}
```

Comment:
Should we use `Schema` in these names, for clarity?

```
// Convert between public schema types and internal data types.
// Avoids coupling public schemas to internal implementation details.
export interface SchemaMapper<Schema, Internal> {
    fromSchema(input: Schema): Internal;
    toSchema(input: Internal): Schema;
}
```

Could also use a separate mapper for things like `CreateStrategySchema` to avoid `mapInput`, I think.

It also looks like all mappers could be static, without the need for `new` instances.

Reply:
sure renaming to include schema makes sense, but I prefer to include the mapInput here as well as it groups same functionality.